android/expoview/src/main/java/host/exp/exponent/notifications/schedulers/IntervalScheduler.java

```diff
@@ -0,0 +1,240 @@
+import host.exp.exponent.notifications.managers.SchedulersManagerProxy;
+import host.exp.exponent.notifications.interfaces.SchedulerInterface;
+import host.exp.exponent.notifications.managers.SchedulersDatabase;
+
+@Table(databaseName = SchedulersDatabase.NAME)
+public class IntervalScheduler extends BaseModel implements SchedulerInterface {
+
+  private List<String> mTriggeringActions = Arrays.asList(null,
+      Intent.ACTION_REBOOT,
+      Intent.ACTION_BOOT_COMPLETED);
+
+  private Context mApplicationContext;
+
+  public HashMap<String, Object> details;
+
+  // -- model fields --
+
+  @Column
+  @PrimaryKey
+  int id;
+
+  @Column
+  int notificationId;
+
+  @Column
+  String experienceId;
+
+  @Column
+  boolean repeat;
+
+  @Column
+  boolean scheduled = false;
+
+  @Column
+  String serializedDetails;
+
+  @Column
+  long scheduledTime;
+
+  @Column
+  long interval;
+
+  // -- scheduler methods --
+
+  @Override
+  public void setApplicationContext(Context context) {
+    mApplicationContext = context.getApplicationContext();
+  }
+
+  @Override
+  public void schedule(String action) throws UnableToScheduleException {
+    if (!mTriggeringActions.contains(action)) {
+      return;
+    }
+    long nextAppearanceTime = 0;
+
+    try {
+      nextAppearanceTime = getNextAppearanceTime();
+    } catch (IllegalArgumentException e) {
+      throw new UnableToScheduleException();
+    }
+
+    ensureDetails();
+    try {
+      getManager().schedule(experienceId, notificationId, details, nextAppearanceTime, null);
+    } catch (ClassNotFoundException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void ensureDetails() {
+    try {
+      details = HashMapSerializer.deserialize(serializedDetails);
+    } catch (JSONException e) {
+      e.printStackTrace();
+    }
+  }
+
+  @Override
+  public void cancel() {
+    getManager().cancel(experienceId, notificationId);
+  }
+
+  @Override
+  public boolean canBeRescheduled() {
+    return repeat || (DateTime.now().toDate().getTime() < scheduledTime);
+  }
+
+  @Override
+  public void onPostSchedule() {
+    this.scheduled = true;
+    save();
+  }
+
+  @Override
+  public String saveAndGetId() {
+    details.put(SchedulersManagerProxy.SCHEDULER_ID, getIdAsString());
+    setDetails(details);
+    save();
+    return getIdAsString();
+  }
+
+  @Override
+  public String getOwnerExperienceId() {
+    return experienceId;
+  }
+
+  @Override
+  public String getIdAsString() {
+    return Integer.valueOf(id).toString();
+  }
+
+  @Override
+  public void remove() {
+    cancel();
+    delete();
+  }
+
+  private long getNextAppearanceTime() { // elapsedTime
+    // time when notification should be presented can be represented as (interval * t + scheduledTime)
+
+    long now = DateTime.now().toDate().getTime();
+    long whenShouldAppear = -1;
+    if (now <= scheduledTime) {
+      whenShouldAppear =  scheduledTime;
+    } else {
+
+      if (interval <= 0) {
```

Comment:
We should validate this argument a lot earlier, maybe in the `NotificationsModule`?